plaso/preprocessors/mediator.py

```diff
@@ -113,11 +113,12 @@ def AddUserAccount(self, user_account):
       KeyError: if the user account already exists.
     """
     if not user_account.username:
-      logger.debug(f'adding user account: {user_account.identifier:s}')
+      logger.debug('adding user account: {0!s}'.format(user_account.username))
```

Comment:
Please use the corresponding f string notation instead

Reply:
username is not set why debug print it?

Reply:
I don't thing the issue is about printing it or not. I thing the f-string is the issue here, it **is** extrapolated whether you pass into logger or not. Hence the bug.

Reply:
so basically, if you want to keep the f-string, enclosing the "debug" statement with an "if" close testing if the values are **not** none would work.

Reply:
can you provide the edge case data why identifier is not set? given it is derived from the Registry value name

Reply:
This is a raw image from an ongoing investigation, I'm afraid I cannot give it. I can run whatever debug command you need though if it can help

Reply:
try running the pre-release and see if the issue still surfaces.

Also no need to provide the "exact" data but something representative enough of the edge case, e.g. change the username to "John Doe", if the identifier is an UUID generate a different one. As long as the example is representative of the edge case.